packages/xrpl/test/models/CredentialDelete.test.ts

```diff
@@ -0,0 +1,171 @@
+import { stringToHex } from '@xrplf/isomorphic/dist/utils'
+import { assert } from 'chai'
+
+import { validate, ValidationError } from '../../src'
+import { validateCredentialDelete } from '../../src/models/transactions/CredentialDelete'
+
+/**
+ * CredentialDelete Transaction Verification Testing.
+ *
+ * Providing runtime verification testing for each specific transaction type.
+ */
+describe('CredentialDelete', function () {
+  let credentialDelete
+
+  beforeEach(function () {
+    credentialDelete = {
+      TransactionType: 'CredentialDelete',
+      Issuer: 'r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ',
+      Subject: 'rNdY9XDnQ4Dr1EgefwU3CBRuAjt3sAutGg',
+      Account: 'rNdY9XDnQ4Dr1EgefwU3CBRuAjt3sAutGg',
+      CredentialType: stringToHex('Passport'),
+      Sequence: 1337,
+      Flags: 0,
+    } as any
+  })
+
+  it(`verifies valid credentialDelete`, function () {
+    assert.doesNotThrow(() => validateCredentialDelete(credentialDelete))
+    assert.doesNotThrow(() => validate(credentialDelete))
+  })
+
+  it(`throws w/ missing field Account`, function () {
+    delete credentialDelete.Account
+    const errorMessage = 'CredentialDelete: missing field Account'
+    assert.throws(
+      () => validateCredentialDelete(credentialDelete),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialDelete),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ Account not string`, function () {
+    credentialDelete.Account = 123
+    const errorMessage = 'CredentialDelete: invalid field Account'
+    assert.throws(
+      () => validateCredentialDelete(credentialDelete),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialDelete),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ Subject not string`, function () {
+    credentialDelete.Subject = 123
+    const errorMessage = 'CredentialDelete: invalid field Subject'
+    assert.throws(
+      () => validateCredentialDelete(credentialDelete),
+      ValidationError,
+      errorMessage,
+    )
+    assert.throws(
+      () => validate(credentialDelete),
+      ValidationError,
+      errorMessage,
+    )
+  })
+
+  it(`throws w/ Issuer not string`, function () {
+    credentialDelete.Issuer = 123
+    const errorMessage = 'CredentialDelete: invalid field Issuer'
```

Comment:
I think we can leave all these as is for now, and then make a refactor in the future to improve the general use helper methods to provide specific typing error message ("expected type XXX, received type YYY") so that this is fixed in all places

Reply:
Yes, please create an issue to track this.